private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/lockfile/V2LockFile.java

```diff
@@ -12,32 +12,175 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package com.github.bazelbuild.rules_jvm_external.coursier;
+package com.github.bazelbuild.rules_jvm_external.resolver.lockfile;
+
+import static com.google.common.base.StandardSystemProperty.USER_HOME;
 
 import com.github.bazelbuild.rules_jvm_external.Coordinates;
+import com.github.bazelbuild.rules_jvm_external.resolver.Conflict;
 import com.github.bazelbuild.rules_jvm_external.resolver.DependencyInfo;
+import com.google.gson.Gson;
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.nio.file.Paths;
 import java.util.Collection;
+import java.util.HashSet;
 import java.util.LinkedHashMap;
 import java.util.LinkedHashSet;
 import java.util.Map;
-import java.util.Objects;
+import java.util.Optional;
 import java.util.Set;
 import java.util.SortedSet;
 import java.util.TreeMap;
 import java.util.TreeSet;
 import java.util.stream.Collectors;
 
 /** Format resolution results into the v2 lock file format. */
-public class NebulaFormat {
+public class V2LockFile {
+
+  private final Collection<URI> allRepos;
+  private final Set<DependencyInfo> infos;
+  private final Set<Conflict> conflicts;
+
+  public V2LockFile(
+      Collection<URI> repositories, Set<DependencyInfo> infos, Set<Conflict> conflicts) {
+    this.allRepos = repositories;
+    this.infos = infos;
+    this.conflicts = conflicts;
+  }
+
+  public Collection<URI> getRepositories() {
+    return allRepos;
+  }
+
+  public Set<DependencyInfo> getDependencyInfos() {
+    return infos;
+  }
+
+  public Set<Conflict> getConflicts() {
+    return conflicts;
+  }
+
+  @SuppressWarnings("unchecked")
+  public static V2LockFile create(String from) {
+    URI m2LocalUri = Paths.get(USER_HOME.value()).resolve(".m2/repository").toUri();
```

Comment:
nit: make this static and share it with `render`

Reply:
Done